nmrfx-structure/src/main/java/org/nmrfx/structure/tools/PeakGenerator.java

```diff
@@ -0,0 +1,230 @@
+                        var peakDim = peak.getPeakDim(0);
+                        peakDim.setLabel(atom.getShortName());
+                        peakDim.setChemShiftValue(atom.getRefPPM().floatValue());
+                        peakDim.setLineWidthHz(1.5f);
+                        peakDim.setBoundsHz(3.0f);
+                        var jCouplings = jCouplingMap.get(atom);
+                        if ((jCouplings != null)) {
+                            List<Double> values = new ArrayList<>();
+                            List<Double> sin2thetas = new ArrayList<>();
+                            List<String> types = new ArrayList<>();
+
+                            var multiplet = peakDim.getMultiplet();
+                            for (var jCoupling : jCouplingMap.get(atom)) {
+                                int nAtoms = jCoupling.getNAtoms();
+                                double value;
+                                if (nAtoms == 3) {
+                                    value = 12.0;
+                                } else {
+                                    value = 5.0;
+                                }
+                                values.add(value);
+                                types.add("d");
+                                sin2thetas.add(0.0);
+                            }
+                            var couplingPattern = new CouplingPattern(multiplet, values, types, sin2thetas, 1.0);
+                            multiplet.setCoupling(couplingPattern);
+                        }
+                    }
+
+                });
+
+    }
+
+
+    public void generateTOCSY(PeakList peakList) {
+        if (tocsyCouplingMap == null) {
+            generateCouplings();
+        }
+        Atom[] atoms = new Atom[2];
+        molecule.getAtoms().stream().
+                filter(atom -> atom.getAtomicNumber() == 1).
+                filter(atom -> !atom.isMethyl() || atom.isFirstInMethyl()).
+                forEach(atom -> {
+                    var tocsyCouplings = tocsyCouplingMap.get(atom);
+                    if ((tocsyCouplings != null)) {
+                        for (var tocsyCoupling : tocsyCouplings) {
+                            atoms[0] = tocsyCoupling.getAtom(0);
+                            atoms[1] = tocsyCoupling.getAtom(1);
+                            double intensity = atom.isMethyl() ? 3.0 : 1.0;
+                            addPeak(peakList, intensity, atoms);
+                        }
+                    }
+                });
+    }
+
+    public void generateHSQC(PeakList peakList, int parentElement) {
+        Atom[] atoms = new Atom[2];
+        molecule.getAtoms().stream().
+                filter(atom -> atom.getAtomicNumber() == 1).
+                filter(atom -> !atom.isMethyl() || atom.isFirstInMethyl()).
+                filter(atom -> atom.getParent() != null && atom.getParent().getAtomicNumber() == parentElement).
+                forEach(atom -> {
```

Comment:
same here if you want to change
```suggestion
        molecule.getAtoms().stream()
                .filter(atom -> atom.getAtomicNumber() == 1)
                .filter(atom -> !atom.isMethyl() || atom.isFirstInMethyl())
                .filter(atom -> atom.getParent() != null && atom.getParent().getAtomicNumber() == parentElement)
                .forEach(atom -> {
```

Reply:
The new method is called from a class in nmrfx-private, so it isn't seen in this repository